### PR TITLE
Scripts for 3858 issue

### PR DESCRIPTION
**Pull request by GetmanetsIrina**
ATF Test Scripts to check #[3858](https://github.com/smartdevicelink/sdl_core/issues/3858)

This PR is **ready** for review.

### Summary
Scripts for 3858 issue

### ATF version
develop

### Changelog
- Added new scripts
- Updated test set for 8_1 release with new added scripts

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by GetmanetsIrina** [comment]
@theresalech, this PR is ready for Livio review.

File: test_scripts/Defects/8_1/3858/common_3858.lua [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+m.preconditions = actions.preconditions
+m.postconditions = actions.postconditions
+m.getMobileSession = actions.getMobileSession
+m.getHMIConnection = actions.getHMIConnection
+m.registerApp = actions.registerApp
+m.activateApp = actions.activateApp
+m.getHMIAppId = actions.app.getHMIId
+
+--[[ Common Variables ]]
+m.tcs = {
```
Comment (ShobhitAd) [inline test_scripts/Defects/8_1/3858/common_3858.lua:21]: I'm able to understand the other abbreviations used here(`CICS`, `PI`) but what does `tcs` stand for?

File: test_scripts/Defects/8_1/3858/3858_1_CreateInteractionChoiceSet_custom_success_codes_result_structure.lua [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+-- during processing of CreateInteractionChoiceSet
+
+-- Precondition:
+-- 1. SDL and HMI are started
+-- 2. Mobile app is registered and activated
+--
+-- Steps:
+-- 1. Mobile app requests CreateInteractionChoiceSet RPC with vrCommands
+-- 2. SDL sends VR.AddCommand(type="Choice") request to the HMI
+-- 3. HMI responds with succeed result code in result structure to VR.AddCommand
```
Comment (ShobhitAd) [inline test_scripts/Defects/8_1/3858/3858_1_CreateInteractionChoiceSet_custom_success_codes_result_structure.lua:14]:
```suggestion
-- 3. HMI responds with success result code in result structure to VR.AddCommand
```

File: test_scripts/Defects/8_1/3858/3858_2_CreateInteractionChoiceSet_custom_success_codes_error_structure.lua [hunk 3]
```diff
@@ -0,0 +1,44 @@
+---------------------------------------------------------------------------------------------------
+-- Issue: https://github.com/smartdevicelink/sdl_core/issues/3858
+---------------------------------------------------------------------------------------------------
+-- Description: Check that SDL transfers succeed result code in error structure from HMI to the mobile app
```
Comment (ShobhitAd) [inline test_scripts/Defects/8_1/3858/3858_2_CreateInteractionChoiceSet_custom_success_codes_error_structure.lua:4]:
```suggestion
-- Description: Check that SDL transfers success result code in error structure from HMI to the mobile app
```

File: test_scripts/Defects/8_1/3858/3858_2_CreateInteractionChoiceSet_custom_success_codes_error_structure.lua [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+-- during processing of CreateInteractionChoiceSet
+
+-- Precondition:
+-- 1. SDL and HMI are started
+-- 2. Mobile app is registered and activated
+--
+-- Steps:
+-- 1. Mobile app requests CreateInteractionChoiceSet RPC with vrCommands
+-- 2. SDL sends VR.AddCommand(type="Choice") request to the HMI
+-- 3. HMI responds with succeed result code in error structure to VR.AddCommand
```
Comment (ShobhitAd) [inline test_scripts/Defects/8_1/3858/3858_2_CreateInteractionChoiceSet_custom_success_codes_error_structure.lua:14]:
```suggestion
-- 3. HMI responds with success result code in error structure to VR.AddCommand
```

**Review by ShobhitAd — CHANGES_REQUESTED** [review]
Please pull changes from the latest develop and move the scripts to 8_2 (https://github.com/smartdevicelink/sdl_atf_test_scripts/tree/develop/test_scripts/Defects/8_2)

File: test_scripts/Defects/8_1/3858/3858_2_CreateInteractionChoiceSet_custom_success_codes_error_structure.lua (hunk 4 above)
Comment (GetmanetsIrina) [reply test_scripts/Defects/8_1/3858/3858_2_CreateInteractionChoiceSet_custom_success_codes_error_structure.lua]: All suggested changes have been applied in [542cf75](https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2648/commits/542cf757f8fa71c65866b619aefaa8e22f323935)

File: test_scripts/Defects/8_1/3858/common_3858.lua (hunk 1 above)
Comment (ShobhitAd) [reply test_scripts/Defects/8_1/3858/common_3858.lua:21]: No, that makes sense. Thanks for clarifying

**Review by ShobhitAd — APPROVED** [review]

**Comment by ShobhitAd** [comment]
@GetmanetsIrina  Approved the tests. Please move the tests to the Defects 8_2 folder and resolve the merge conflicts

**Pull request merged**